File: webapi/Models/HR/Insurance.cs
using System.ComponentModel.DataAnnotations;
using webapi.Models.System;

namespace webapi.Models.HR
{
    public class Insurance
    {
        [Key]
        public int Id { get; set; }
        public decimal DooWithouthTzpbInsurer { get; set; }
        public decimal DooWithouthTzpbEmployee { get; set; }
        public decimal HealthInsurance { get; set; }
        public decimal HealthInsuranceArticle40 { get; set; }
        public decimal HealthInsuranceInsurer { get; set; }
        public decimal HealthInsuranceEmployee { get; set; }
        public decimal TeacherPensionFund { get; set; }
        public decimal ProfessionalPensionFund { get; set; }
        public decimal UniversalPensionInsurer { get; set; }
        public decimal UniversalPensionEmployee { get; set; }
        public int InsuranceTypeId { get; set; }
        public SysInsuranceType? SysInsuranceTypeId { get; set; }
    }


    public class InsuranceDTO
    {
        public decimal DooWithouthTzpbInsurer { get; set; }
        public decimal DooWithouthTzpbEmployee { get; set; }
        public decimal HealthInsurance { get; set; }
        public decimal HealthInsuranceArticle40 { get; set; }
        public decimal HealthInsuranceInsurer { get; set; }
        public decimal HealthInsuranceEmployee { get; set; }
        public decimal TeacherPensionFund { get; set; }
        public decimal ProfessionalPensionFund { get; set; }
        public decimal UniversalPensionInsurer { get; set; }
        public decimal UniversalPensionEmployee { get; set; }
        public int InsuranceTypeId { get; set; }
    }
}

File: reactapp/src/pages/Employees/models/Insurance.ts
export interface Insurance {
    id: int
    dooWithouthTzpbInsurer: string,
    dooWithouthTzpbEmployee: string,
    healthInsurance: string,
    healthInsuranceArticle40: string,
    healthInsuranceInsurer: string,
    healthInsuranceEmployee: string,
    teacherPensionFund: string,
    professionalPensionFund: string,
    universalPensionInsurer: string,
    universalPensionEmployee: string,
    insuranceTypeId: int,
}

export const defaultInsuranceUpdateDTO: Insurance = {
    id: 0,
    dooWithouthTzpbInsurer: "",
    dooWithouthTzpbEmployee: "",
    healthInsurance: "",
    healthInsuranceArticle40: "",
    healthInsuranceInsurer: "",
    healthInsuranceEmployee: "",
    teacherPensionFund: "",
    professionalPensionFund: "",
    universalPensionInsurer: "",
    universalPensionEmployee: "",
    insuranceTypeId: 0,
}

export interface InsuranceDTO {
    dooWithouthTzpbInsurer: string,
    dooWithouthTzpbEmployee: string,
    healthInsurance: string,
    healthInsuranceArticle40: string,
    healthInsuranceInsurer: string,
    healthInsuranceEmployee: string,
    teacherPensionFund: string,
    professionalPensionFund: string,
    universalPensionInsurer: string,
    universalPensionEmployee: string,
    insuranceTypeId: int,
}


export const defaultInsuranceDTO: InsuranceDTO = {
    dooWithouthTzpbInsurer: "",
    dooWithouthTzpbEmployee: "",
    healthInsurance: "",
    healthInsuranceArticle40: "",
    healthInsuranceInsurer: "",
    healthInsuranceEmployee: "",
    teacherPensionFund: "",
    professionalPensionFund: "",
    universalPensionInsurer: "",
    universalPensionEmployee: "",
    insuranceTypeId: 0,
}



export interface InsuranceUpdateData {
    insuranceTypeId: string,
}

export const defaultInsuranceUpdateData: InsuranceUpdateData = {
    insuranceTypeId: "",
}

export function createInsuranceUpdateData(code: string): InsuranceUpdateData {
    return {
        insuranceTypeId: code,
    }
}